source/fundamentals/databases-collections.txt

```diff
@@ -0,0 +1,185 @@
+a ``MongoClient`` instance to access a ``MongoDatabase`` in a MongoDB
+instance.
+
+The following example accesses a database named "testDatabase":
+
+.. code-block:: java
+
+   MongoDatabase database = mongoClient.getDatabase("testDatabase");
+
+Access a Collection
+-------------------
+
+Use the :java-sync-api:`getCollection()
+<com/mongodb/client/MongoDatabase.html#getCollection(java.lang.String)>`
+method of a ``MongoDatabase`` instance to access a
+``MongoCollection`` in a database of your connected MongoDB instance.
+
+The following example accesses a collection named "testCollection" from
+a ``MongoDatabase``:
+
+.. code-block:: java
+
+   MongoDatabase database = mongoClient.getDatabase("testDatabase");
+   MongoCollection<Document> collection = database.getCollection("testCollection");
+
+.. note::
+
+   If the provided collection name does not already exist in the database,
+   MongoDB implicitly creates the collection when you first insert data
+   into that collection.
+
+Create a Collection
+-------------------
+
+Use the :java-sync-api:`createCollection()
+<com/mongodb/client/MongoDatabase.html#createCollection(java.lang.String,com.mongodb.client.model.CreateCollectionOptions)>`
+method of a ``MongoDatabase`` instance to create a collection called
+"exampleCollection" in a database of your connected MongoDB instance:
+
+.. code-block:: java
+
+   database.createCollection("exampleCollection");
+
+You can specify collection options like maximum size and document
+validation rules using the :java-docs:`CreateCollectionOptions
+<apidocs/mongodb-driver-core/com/mongodb/client/model/CreateCollectionOptions.html>`
+class. ``createCollection()`` accepts an instance of
+``CreateCollectionOptions`` as an optional second parameter.
+
+Capped Collection
+~~~~~~~~~~~~~~~~~
+
+A **capped collection** is a collection with a preset maximum size.
+To create a capped collection, use the :java-docs:`CreateCollectionOptions.capped()
+<apidocs/mongodb-driver-core/com/mongodb/client/model/CreateCollectionOptions.html?is-external=true#capped(boolean)>`
+method with a specified size:
+
+.. code-block:: java
+
+   database.createCollection("exampleCappedCollection",
+       new CreateCollectionOptions().capped(true).sizeInBytes(0x100000));
+
+For more information, see the server documentation for :manual:`capped
+collections </core/capped-collections>`.
+
+Document Validation
+~~~~~~~~~~~~~~~~~~~
+
+**Document validation** provides the ability to validate documents
+against a series of filters during writes to a collection. You can
+specify these filters using the :java-docs:`ValidationOptions
+<apidocs/mongodb-driver-core/com/mongodb/client/model/ValidationOptions.html>`
+class, which accepts a series of :java-docs:`Filters
+<apidocs/mongodb-driver-core/com/mongodb/client/model/Filters.html>`
+that specifies the validation rules and expressions:
+
+.. code-block:: java
+
+   ValidationOptions collOptions = new ValidationOptions().validator(
```

Comment:
Should we instead be pushing JSON schema validation?

Reply:
For most use cases, I'd definitely go with JSON schema. However, there doesn't seem to be any easy way to create a JSON schema in Java without manually creating a bunch of nested Document objects, which seems very error prone. Is there some better way? Otherwise, I'd vote that we should leave the Filters example in place because it's much more intuitive and works just as well for many users.

Reply:
`BsonDocument.parse(<json string>`) is an easy way to do it.  Up to you, though.

Reply:
Going to leave it as is so we don't overcomplicate the example for new users. If we receive feedback I'm sure we can always create a JSON Schema page or section in the future.